src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
Re 2.: The KLSM doesn't have an internal retry loop, it only retries once after spying if the first attempt failed IIRC. It just seems weird to me that some delete_min's have a retry loop and others don't.